projects/packages/identity-crisis/src/class-identity-crisis.php

```diff
@@ -1394,4 +1395,65 @@ public static function register_request_body( array $params ) {
 	public static function site_registered( $blog_id ) {
 		update_option( static::PERSISTENT_BLOG_ID_OPTION_NAME, (int) $blog_id, false );
 	}
+
+	/**
+	 * If URL is an IP, add the IP value to the ip_requester option with its expiry value.
+	 *
+	 * @throws Exception Thrown if unable to save the new IP requester.
+	 */
+	public static function set_ip_requester_for_idc() {
+		// Check if option exists
+		$data = Jetpack_Options::get_option( 'identity_crisis_ip_requester' );
+		// If not set it
+		$site_ip      = wp_parse_url( Urls::site_url(), PHP_URL_HOST );
+		$ip_requester = array(
+			'ip'         => $site_ip,
+			'expires_at' => time() + 300,
+		);
+		if ( ! $data && ! empty( $ip_requester ) ) {
+			$data[] = $ip_requester;
+			Jetpack_Options::update_option( 'identity_crisis_ip_requester', $data );
+		} else {
+			$updated_data  = array();
+			$updated_value = false;
+			foreach ( $data as $item ) {
+				// If set, delete any expired values first
+				if ( time() > $item['expires_at'] ) {
+					$item = null;
+				}
+				if ( $item ) {
+					if ( $item['ip'] === $site_ip ) {
+						$item['expires_at'] = time() + 300;
+						$updated_value      = true;
+					}
+					$updated_data[] = $item;
+				}
+			}
+			if ( ! $updated_value ) {
+				$updated_data[] = $ip_requester;
+			}
+			if ( $data !== $updated_data ) {
+				Jetpack_Options::update_option( 'identity_crisis_ip_requester', $updated_data );
```

Comment:
It may be possible with sites that use multiple IPs and have heavy traffic. I'm just not sure how likely is that to happen. We could add some locking system or queuing but I worry we could be overcomplicating things that will never turn out to be a real problem.

Reply:
I agree, let's not complicate it any further than we have to 🙂